### ISSUE TITLE
Exiting a yazi spawned shell in fish, exits yazi instead of the spawned shell.

### ISSUE THREAD

**Issue by ThGrSoRu** [sxyazi/yazi#432]
### What system are you running Yazi on?

Linux Wayland

### What terminal are you running Yazi in?

foot

### Yazi version

yazi 0.1.5 (dc1649f 2023-12-04)

### Did you try the latest main branch to see if the problem has already been fixed?

Tried, but the problem is still present

### Describe the bug

Dropping into a fish shell from within yazi using: `{on=["T"],exec="shell fish --block --confirm"},`, and typing `exit` in that shell shows: `fish: Job 1, 'yazi' has stopped` and does not exit the shell.
The yazi process now shows up in htop with a T as status, meaning it has been stopped by a control signal.
Typing `exit` again shows: `There are still jobs active:
6638 yazi`
Typing `exit` for the third time actually quits the shell, leaving the yazi process as a single orphan consuming one cpu core worth, only able to be shut down using a `SIGKILL`.

Typing `exit` in a fish shell not spawned by yazi, exits the shell instantly.

### Expected Behavior

The shell should be quit, not yazi.

### To Reproduce

See above.

**Comment by ThGrSoRu** [comment]
Opening a new terminal window running fish, chaning to bash, then running yazi from that bash shell, then dropping into a fish shell from yazi, then exiting back into the first fish shell shows the same behaviour but also gives an error message:
```
Backtrace (most recent call first):
  File "<unknown>", line 0, in __libc_start_main

The application panicked (crashed).
  called `Result::unwrap()` on an `Err` value: Input/output error (os error 5)
in yazi-fm/src/app/commands/stop.rs, line 32
thread: main
```
In the report above, an error message is never displayed.
There is also no orphan yazi process left running.

**Comment by sxyazi** [comment]
I can't reproduce it on my side, with fish 3.6.4. Could you record a video? I'd like to see your steps.

https://github.com/sxyazi/yazi/assets/17523360/cc9356cb-a194-44ad-9537-bba14a23f955

**Comment by sxyazi** [comment]
I'm trying to install fish from chaotic-aur, but I can't update the source. I'm setting it up on a fresh Arch Linux following the ["How to use it"](https://aur.chaotic.cx/) instructions from chaotic-aur:

```bash
pacman-key --recv-key 3056513887B78AEB --keyserver keyserver.ubuntu.com
pacman-key --lsign-key 3056513887B78AEB
pacman -U 'https://cdn-mirror.chaotic.cx/chaotic-aur/chaotic-keyring.pkg.tar.zst'
pacman -U 'https://cdn-mirror.chaotic.cx/chaotic-aur/chaotic-mirrorlist.pkg.tar.zst'
```

And:

```ini
#/etc/pacman.conf
[chaotic-aur]
Include = /etc/pacman.d/chaotic-mirrorlist
```

![x](https://github.com/sxyazi/yazi/assets/17523360/d87428d2-f2f5-4934-8104-c7815f7d5c9e)

I want to know if there's any alternative method besides chaotic-aur to accomplish this?

**Comment by ThGrSoRu** [comment]
I also did
```
pacman-key --recv-key 3056513887B78AEB --keyserver keyserver.ubuntu.com
pacman-key --lsign-key 3056513887B78AEB
```
but I only put
```
[chaotic-aur]
Server = https://geo-mirror.chaotic.cx/$repo/$arch
```
into pacman.conf.

Beyond that, it seems a lot of mirrors are down right now.
Although `https://geo-mirror.chaotic.cx/$repo/$arch` has always worked for me.

**Comment by ThGrSoRu** [comment]
This issue seems to be gone using fish 6a64ba6 and yazi 93c8d90.

**Comment by ThGrSoRu** [comment]
This issue seems to be resolved with fish 6a64ba6 and yazi 93c8d90.
